### PR TITLE
feat(typescript-axios): add explicit override modifier

### PR DESCRIPTION
**Pull request by dschu-lab**
### Introducing the following changes
- Updated minimum typescript version to `^4.3`
- Added `override` modifier to base api template
- Updated samples

### Motivation
The setting `noImplicitOverride` leads to a more robust codebase. Even if the option is not activated, the class still works if at least TS 4.3 is installed.

### References

- [TS Playground running v4.3.5 with `noImplicitOverride ` set to `false´](https://www.typescriptlang.org/play?noImplicitReturns=false&ts=4.3.5#code/KYDwDg9gTgLgBAYwDYEMDOa4CVgEcCuAllMACYCiUU0coMwAdqZpdVHAN4BQcvcEAN2BVCpYHAYoAtsABccAEQ4CxMq2gK4AXkXKiJClQ0BuHnwQQGaGFHwIY0ABRh8AIySEEcAGaFgSUnlrKEIGAHMAGjgpNDCAfiCbULCASk4zPj40fDBhRxjU00yAXy5ioA)
- [TS Documentation noImplicitOverride](https://www.typescriptlang.org/tsconfig#noImplicitOverride)

### Pinging the commitee for visibility
@TiFu @taxpon @sebastianhaas @kenisteward @Vrolijkx  @macjohnny  @topce @akehir  @petejohansonxo @amakhrov  @davidgamero  @mkusaka 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: samples/client/others/typescript-axios/with-separate-models-and-api-inheritance/common.ts [hunk 1]
```diff
@@ -85,6 +85,7 @@ export const setOAuthToObject = async function (object: any, name: string, scope
 }
 
 function setFlattenedQueryParams(urlSearchParams: URLSearchParams, parameter: any, key: string = ""): void {
+    if (parameter == null) return;
```
Comment (amakhrov) [inline samples/client/others/typescript-axios/with-separate-models-and-api-inheritance/common.ts:88]: where does this come from?

**Review by amakhrov — APPROVED** [review]
Thanks for the PR! Looks good to me.
Since technically it's a breaking change - should we target the 7.x branch?

Also it looks like we could avoid the problem in the first place by initializing `this.name` in the constructor:
```
export class RequiredError extends Error {
  constructor(...) {
    this.name = "RequiredError"
  }
}
```
This way it would not be sensitive to a TS version :)

File: samples/client/others/typescript-axios/with-separate-models-and-api-inheritance/common.ts (hunk 1 above)
Comment (dschu-lab) [reply samples/client/others/typescript-axios/with-separate-models-and-api-inheritance/common.ts:88]: I've also noticed this. I just ran the commands mentioned in the PR template:
```sh
./mvnw clean package 
./bin/generate-samples.sh
./bin/utils/export_docs_generators.sh
```

**Comment by dschu-lab** [comment]
> Also it looks like we could avoid the problem in the first place by initializing `this.name` in the constructor:
> [...]
> This way it would not be sensitive to a TS version :)

Interesting suggestion! Should I update the PR then? 😄

**Comment by amakhrov** [comment]
> interesting suggestion! Should I update the PR then? 

honestly, if feels like a safer approach, without sacrificing any functionality / ease of usage

**Comment by dschu-lab** [comment]
> > interesting suggestion! Should I update the PR then?
> 
> honestly, if feels like a safer approach, without sacrificing any functionality / ease of usage

Closed this PR and created a new one for this
https://github.com/OpenAPITools/openapi-generator/pull/14230

**Comment by amakhrov** [comment]
Thanks for doing this  @dschu-lab !  and sorry for back-n-forth with the direction :)

**Comment by dschu-lab** [comment]
> Thanks for doing this @dschu-lab ! and sorry for back-n-forth with the direction :)
No problem at all. Ultimately the new implementation is even better because it is backward compatible. 😄 🥂

**Pull request closed without merging**